test/driver.cpp

```diff
@@ -83,6 +87,7 @@ void processCmdLine(std::vector<std::string_view> arguments)
        setPassReportMode(passReportMode::none);
    else if (passReportMode == "auto")
```

Comment:
Consider the following revisions:
- change `passReportMode` variable's type to the enum type of that name
- Refactor the code in L81-L87 to a new function `getPassReportMode` that accepts a `string_view` and returns `passReportMode` enum
- Use function `getPassReportMode` in L65

Reply:
@smurthys I'm not sure how we would resolve when "auto" is supplied as the pass report mode as this requires knowing which file open mode the user has supplied as well which we have no way of knowing for sure until after all the command-line arguments have been parsed.

Perhaps I misread your suggestions but if not an alternative could be to keep `passReportMode` as a `string_view`, refactor the code in L93-L107 into a function `getPassReportMode` that takes in a reference to `passReportMode` and a reference to `fileOpenMode` and returns a `passReportMode` enum. Then use the return value of `getPassReportMode` as the parameter in `setPassReportMode`. I don't see much benefit to this, however, as this is essentially just taking a block of a code and moving it to a function that will only be used once. Perhaps it would increase readability.

Reply:
@Ndemco Maybe I'm missing something but I don't see any issue with using the suggested function on L65 as I propose. Any prm the uses explicitly proposes is used regardless of the what that `-f*` option says. `auto` is the default prm no matter if/what `-f*` option is used. That is, using the value determined on L65 in place of the current code in L81-L87 shouldn't be a problem. The logic starting from L88 about prm being `auto` can still remain except that check will start the `if else` flow.

Having said this, I worry I'm missing something fundamental. I can try making the change if you like and perhaps that would help me see the error in my logic.

Reply:
@smurthys There is no `passReportMode::auto`, so in the `getPassReportMode` function you proposed, if the `string_view` parameter passed to the `getPassReportMode` function is `"auto"`, in order for the function to return a `passReportMode` enum, it would have to know whether the user has chosen to send the results to a file or not by checking if `fileOpenMode` is empty. At first I thought this could be solved by including a reference to `fileOpenMode` in the function parameters, but after thinking on it, I realized that the wrong value for `fileOpenMode` will be passed to `getPassReportMode` if the `-f` option is read after the `-p` option.

Another issue I see is that because there is no `passReportMode::auto`, if the variable `passReportMode`  was set to the enum type of that name, rather than a `string_view` as it is now, we would not be able to give it a default value of `auto`.

It is possible I'm still misunderstanding; if you believe so, it might be best for you to try to make the change so both of us can see where the misunderstanding is.

Reply:
Thanks @Ndemco for helping me see the issue. I had completely missed that `auto` is not one of the enumerated values, and we should not add that as a value. 

I still wrote the function `getPassReportMode` to convert text prm to enum, and used the new function. Please review my changes. They are in commit adca120

Reply:
@smurthys  The changes look great! Can the parameter `const std::string_view value` not be a reference to a `string_view`? (ie. `const std::string_view& value`)

Reply:
Thanks @Ndemco for spotting the omission. I just pushed the changes to address the issue. Please review.

Reply:
I reviewed the code around setting prm and changed fn. `getPassReportMode` to handle auto based on whether or not the output goes to a file. This cleans up the code in `processCmdLine` and enables us to refactor `processCmdLine`. Also in `processCmdLine`, the prm variable is default initialized so accurately reflect the state. `getPassReportMode` interprets empty prm value as "auto" consistent to the documentation.

Reply:
Looks good. Should we initialize `string_view passReportMode` to `"auto"` rather than keeping it empty? I understand you've written `replace_all` to treat an empty string as `auto`, but perhaps initializing `passReportMode` to `"auto"` makes it more explicit that the `auto` option is the default option. We could still leave the code that treats an empty `string_view` as `auto` for the sake of defensive programming.

Reply:
I default-initialized the prm variable intentionally, but I see your point. Please make a suggestion in a comment and I will accept it.

PS: I assume you meant function `getPassReportMode` instead of `replace_all` as treating empty prm value as "auto".

Reply:
Yes, my mistake! I meant `getPassReportMode`.